### PR TITLE
Updated date_detection Module to return minimum & maximum  dates  for each date column

### PR DESCRIPTION
**Pull request by bhagyakjain**
Main Changes -
1. Updated date_detection Module to return minimum & maximum  dates  for each date column
2. Modified the expected results in the tests
3. Corrected a spelling mistake

Also Added -
1. main.py & requirements.txt the files required for deploying to the cloud endpoint
2. debug.py - script that minics the cloud endpoint - so helps in deploying 
3. obj.json - file where json from UI is pasted before running debug.py 
4. test_all.sh to run all tests at a time & showing results

### CODE REVIEW

File: date_detection/main.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+"""
+"""
+This module contains the date detection function that gets called by the UI.
+It returns a json containing the dict of all the detected date columns.
+"""
+import date_detection
+from util import enums
+import pandas, json
+
+def hello_world(request):
```
Comment (arijitde-goog) [inline date_detection/main.py:23]: Why is this called hello_world?

File: date_detection/debug.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+"""
+This module will help in calling the main.py locally and debugging
+if any error occurs.
+This module has a RequestClass that mimics the flask request class.
```
Comment (arijitde-goog) [inline date_detection/debug.py:18]: What is "flask"?

File: date_detection/main.py (hunk 1 above)
Comment (bhagyakjain) [reply date_detection/main.py:26]: Added a todo

File: date_detection/debug.py (hunk 2 above)
Comment (bhagyakjain) [reply date_detection/debug.py]: The hello_world function gets a <request (flask.Request): HTTP request object>  argument when it is called by the UI.

As it was difficult to create an actual flask request locally, we chose to create a dummy request using our own class.

**Comment by bhagyakjain** [comment]
> No tests for the new min / max logic?

Added tests.

**Pull request merged**